### ISSUE TITLE
arithmetic_side_effects should not report UNIX_EPOCH + duration

### ISSUE THREAD

**Issue by software-artificer** [rust-lang/rust-clippy#17489]
### Summary

At the moment, clippy reports `arithmetic operation that can potentially result in unexpected side-effects` when performing `std::time::UNIX_EPOCH + std::time::Duration`. Because `Duration` is represented as `u64` and `UNIX_EPOCH` is zero, we know that overflow simply cannot occur in this scenario, no matter what.

If everyone agrees that this is a good idea, I am happy to try and implement this as well.

### Lint Name

arithmetic_side_effects

### Reproducer

I tried this code:

```rust
fn main() {
    std::time::UNIX_EPOCH + std::time::Duration::from_secs(1);
}
```

I saw this happen:

```
error: arithmetic operation that can potentially result in unexpected side-effects
 --> src/main.rs:2:5
  |
2 |     std::time::UNIX_EPOCH + std::time::Duration::from_secs(1);
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#arithmetic_side_effects
  = note: requested on the command line with `-D clippy::arithmetic-side-effects`
```

I expected to see this happen:
No error should be reported by Clippy.

### Version

```text
rustc 1.97.0 (2d8144b78 2026-07-07) (built from a source tarball)
binary: rustc
commit-hash: 2d8144b7880597b6e6d3dfd63a9a9efae3f533d3
commit-date: 2026-07-07
host: aarch64-apple-darwin
release: 1.97.0
LLVM version: 21.1.8
```

**Comment by Jarcho** [comment]
`UNIX_EPOCH` is only zero on some platforms (e.g. on Windows it's `11_644_473_600 * INTERVALS_PER_SEC as i64`). If the `Add` trait were const and clippy had some very non-trivial const eval extensions we could catch your reproducer not panicking, but that's not worth leaving an issue open.

**Comment by software-artificer** [comment]
@Jarcho ah of course, keep forgetting Windows exists 😆
